pkg/codegen/schema/schema.go

```diff
@@ -607,21 +637,33 @@ type PropertySpec struct {
 	Secret bool `json:"secret,omitempty"`
 }
 
-// ObjectTypeSpec is the serializable form of an object type.
-type ObjectTypeSpec struct {
+// ComplexTypeSpec is the serializable form of an object or enum type.
+type ComplexTypeSpec struct {
```

Comment:
I don't think that we should rename this type. Instead, I think that we should define a new type that embeds an `ObjectTypeSpec` and adds the enum-related fields.

Reply:
For example:

```go
type ComplexTypeSpec struct {
    ObjectTypeSpec

    Enum []EnumValueSpec `json:"enum,omitempty"`
}
```

Then we can define the schema with stronger typing.